airflow/providers/amazon/aws/hooks/eks.py

```diff
@@ -0,0 +1,461 @@
+
+        return self._list_all(api_call=api_call, response_key="clusters", verbose=verbose)
+
+    def list_nodegroups(
+        self,
+        clusterName: str,
+        verbose: Optional[bool] = False,
+    ) -> List:
+        """
+        Lists all Amazon EKS Nodegroups associated with the specified cluster.
+
+        :param clusterName: The name of the Amazon EKS Cluster containing nodegroups to list.
+        :type clusterName: str
+        :param verbose: Provides additional logging if set to True.  Defaults to False.
+        :type verbose: bool
+
+        :return: A List of nodegroup names within the given cluster.
+        :rtype: List
+        """
+        eks_client = self.conn
+        api_call = partial(eks_client.list_nodegroups, clusterName=clusterName)
+
+        return self._list_all(api_call=api_call, response_key="nodegroups", verbose=verbose)
+
+    def _list_all(self, api_call: Callable, response_key: str, verbose: bool) -> List:
+        """
+        Repeatedly calls a provided boto3 API Callable and collates the responses into a List.
+
+        :param api_call: The api command to execute.
+        :type api_call: Callable
+        :param response_key: Which dict key to collect into the final list.
+        :type response_key: str
+        :param verbose: Provides additional logging if set to True.  Defaults to False.
+        :type verbose: bool
+
+        :return: A List of the combined results of the provided API call.
+        :rtype: List
+        """
+        name_collection = []
+        token = DEFAULT_PAGINATION_TOKEN
+
+        try:
+            while token != "null":
+                response = api_call(nextToken=token)
+                # If response list is not empty, append it to the running list.
+                name_collection += filter(None, response.get(response_key))
+                token = response.get("nextToken")
+
+            self.log.info("Retrieved list of %s %s.", len(name_collection), response_key)
+            if verbose:
+                self.log.info("%s found: %s", response_key.title(), name_collection)
+
+        except ClientError as e:
+            self.log.error(e.response["Error"]["Message"])
+            raise e
+        return name_collection
+
+
+def generate_config_file(
+    eks_cluster_name: str,
+    eks_namespace_name: str,
+    pod_name: str,
+    aws_profile: Optional[str],
+    pod_username: Optional[str] = DEFAULT_POD_USERNAME,
+    pod_context: Optional[str] = DEFAULT_CONTEXT_NAME,
+    aws_region: Optional[str] = None,
+) -> str:
+    """
+    Writes the kubeconfig file given an EKS Cluster.
+
+    :param eks_cluster_name: The name of the cluster to create the EKS Managed Nodegroup in.
+    :type eks_cluster_name: str
+    :param eks_namespace_name: The namespace to run within kubernetes.
+    :type eks_namespace_name: str
+    :param pod_name: The unique name to give the pod.  Used as an identifier in the config filename.
```

Comment:
Can we generate this pod name for each run?

Reply:
Absolutely, that would be easy enough.   Is there no use case where the user would want to know the pod name?

Reply:
Accessing the pod name in the operator makes sense, but it doesn't make sense to strictly associate the configuration file with a specific pod name. One configuration file can be used to create multiple pods or perform other operations, such as creating a secret.

Reply:
You can keep this argument if you want, but it should have a generic name like `tmpfile_prefix` and be optional. Most users don't pay much attention to names for temporary files, so we shouldn't force it to pass a value for this argument.

Reply:
If the file is being deleted as you requested, then this isn't really relevant anymore.  I'll change this to just use the default NamedTemporaryFile naming convention instead of overwriting that.

Reply:
Should be addressed by https://github.com/apache/airflow/pull/16571/commits/8687b736994d1911011dd8bd2b2903d82ad30a8c